src/eureka/S5_lightcurve_fitting/differentiable_models/StarryModel.py

```diff
@@ -114,8 +114,80 @@ def setup(self, newparams):
             pl_params = PlanetParams(self, 0, chan, eval=False)
 
             # Initialize star object
-            star = starry.Primary(starry.Map(udeg=self.udeg),
-                                  m=0, r=pl_params.Rs)
+            # check for spots and set parameters
+            if hasattr(self.parameters, 'spotrad0'):
+                # find number of spots
+                spotinds = [s for s in self.parameters.dict.keys() 
+                            if 'spotrad' in s]   
+                counter = [s for s in spotinds if '_' in s]
+                nspots = len(spotinds) - len(counter)
+
+                # create arrays to hold values
+                spotrad = np.zeros((self.nchannel_fitted, nspots))
+                spotlat = np.zeros((self.nchannel_fitted, nspots))
+                spotlon = np.zeros((self.nchannel_fitted, nspots))
+                spotcon = np.zeros((self.nchannel_fitted, nspots))
+                starrot = np.zeros((self.nchannel_fitted))
+                starinc = np.zeros((self.nchannel_fitted))
```

Comment:
At this point in the code, the "values" are theano tensors (since you're pre-compiling things here) and you cannot set elements of a numpy float array with theano tensors. You'll need to instead use `tt.zeros`. But in the `update` function down below, the "values" are floats and you can stick with `np.zeros` there